src/fn/create-users.js

```diff
@@ -18,9 +18,8 @@ module.exports = (projectDir, couchUrl) => {
         const username = row[cols.indexOf('username')];
         const password = row[cols.indexOf('password')];
         const type     = row[cols.indexOf('type')];
-
+        const place    = cols.find(e => e.includes("place.")) ? prefixedProperties(cols, row, 'place.'  ) : row[cols.findIndex(x => x === 'place')];
```

Comment:
This line is a bit complicated for me, and really long! I'm not a big fan of single letter variables although this is a personal preference and up for debate. Finally, I think it'd be easier to read if it was a simple `if/else` rather than a ternary.

In addition, running `cols.find` for each row is inefficient. The other code in this section should also be optimised (eg: `cols.indexOf('type')`). I don't know how much difference this will make in the real world, but it would probably be noticeable if uploading thousands of users.

So how about replacing it with something like this...

```js
...
const { cols, rows } = fs.readCsv(csvPath);
const usernameIndex = cols.indexOf('username');
const passwordIndex = cols.indexOf('password');
const typeIndex = cols.indexOf('type');
const placeIdIndex = cols.indexOf('place'); // may be not found
const contactPropertiesFn = prefixedProperties(cols, 'contact.');
const placePropertiesFn = prefixedProperties(cols, 'place.');
return rows.reduce((promiseChain, row) => {
  const username = row[usernameIndex];
  const password = row[passwordIndex];
  const type = row[typeIndex];
  const contact = contactPropertiesFn(row);
  const place = placeIdIndex ? row[placeIdIndex] : placePropertiesFn(row);
  
  ...
});

// NB this is updated too!
const prefixedProperties = (cols, prefix) => {
  const indices = {};
  cols.forEach(col => {
    if (col.startsWith(prefix)) {
      indices[col.substring(prefix.length)] = cols.indexOf(col);
    }
  });
  return (row) => {
    const result = {};
    Object.keys(indices).forEach(key => result[key] = row[indices[key]]);
    return result;
  };
};
```

I haven't tested any of this so it's probably buggy, and it could be further improved but hopefully it shows what I mean.